lib/ansible/modules/net_tools/nmcli.py

```diff
@@ -1182,6 +1182,17 @@ def modify_connection(self):
             cmd = self.modify_connection_bridge_slave()
         elif self.type == 'vlan':
             cmd = self.modify_connection_vlan()
+            # If no dns given, then we assume that IPv4/IPv6 is not used on the interface
+            # so we are going to disable the protocol, otherwise interface won't come up.
+
+            if self.ip4 is 'disabled':
```

Comment:
You shouldn't do string comparison with `is`.

https://stackoverflow.com/questions/1504717/why-does-comparing-strings-in-python-using-either-or-is-sometimes-produce